test/realm_test.dart

```diff
@@ -1073,5 +1069,29 @@ Future<void> main([List<String>? args]) async {
       expect(File(config.path).existsSync(), true);
       expect(Directory("${config.path}.management").existsSync(), true);
     });
+
+    test('Equals', () {
+      var config = Configuration([Dog.schema, Person.schema]);
+      var realm = Realm(config);
+
+      final person = Person()..name = 'Kasper';
+      final dog = Dog()..name = 'Fido'..owner = person;
+
+      expect(person, person);
+      expect(person, isNot(1)); 
+      expect(person, isNot(dog));
+
+      realm.write(() {
+        realm..add(person)..add(dog);
+      });
+
+      expect(person, person);
+      expect(person, isNot(1)); 
+      expect(person, isNot(dog));
+
+      final read = realm.query<Person>("name == 'Kasper'");
+
+      expect(read, [person]);
```

Comment:
Minor: Not sure what the error message when comparing collections, I would have written `expect(read[0], person);` Hope it's readable.

Reply:
consider ignoring this if you think its alright

Reply:
I think it is a bit safer like this. If read is empty, read[0] throws before expect gets to run. Also, the output is quite nice, but we should consider generating a toString method for realm objects.

Reply:
Also `expect(read[0], person)` don't check if `read.length > 1`

Reply:
sure, scratch my comment

Reply:
scratch my comment